### ISSUE TITLE
[BUG] `check_env_specs` shouldn't set `torch.manual_seed` by default.

### ISSUE THREAD

**Issue by maximilianigl** [pytorch/rl#1855]
## Describe the bug

Current default behaviour of `check_env_specs()` is to set the manual key to 0. 
I think this is dangerous as people might run this function after setting their own seed (at the beginning of `main()`, not realizing it's overwriting what they set before. 

## Reason and Possible fixes

Set default argument of `seed` in `check_env_specs` to `None`.

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
- [x] I have read the [documentation](https://github.com/pytorch/rl/tree/main/docs/) (**required**)
- [ ] I have provided a minimal working example to reproduce the bug (**required**)

**Comment by vmoens** [comment]
Yep I agree it isn't ideal.

Note that the intended use of this function is to be run out of your training loop.

I wish PyTorch had some more refined way of locally set the seed but as of now it isn't the case.

I can have a look at not seeding things by default

**Comment by vmoens** [comment]
If you want to have a look at #1872 it should be ready

**Comment by maximilianigl** [comment]
Looks great, thank you!
